pkg/download/download.go

```diff
@@ -79,7 +101,12 @@ func download(client *http.Client, u *url.URL, path string, extract bool) error
 		}
 		defer resp.Body.Close()
 
-		if resp.StatusCode != http.StatusOK {
+		switch {
+		case resp.StatusCode == http.StatusOK:
+			// OK
+		case resp.StatusCode == http.StatusNotFound && do.dontRetryOn404:
+			return &backoff.PermanentError{Err: NotFound}
```

Comment:
Now that we've moved Ubuntu Kinetic to somewhere that won't periodically 404, we can switch this to erroring all the time, so we don't need this to be configurable, which should clean this code up a lot.

Reply:
I didn't know the retries were there just for the Kinetic 404s, thanks for catching that!

Done in [Remove retries in download now that we've fixed the issue with Ubuntu…](https://github.com/fleetdm/fleet/pull/21749/commits/5a0154f58c256b9d24b123e98c2a857dacc7ac4a).